src/utilities/createCurrentRoute.ts

```diff
@@ -1,22 +1,24 @@
 import { readonly, reactive } from 'vue'
-import { ResolvedRoute } from '@/types/resolved'
+import { ResolvedRoute, RouterRoute } from '@/types/resolved'
 
 type ResolvedRouteUpdate = (route: ResolvedRoute) => void
 
-type CurrentRoute = {
-  route: ResolvedRoute,
+type CurrentRouteContext = {
+  resolvedRoute: ResolvedRoute,
```

Comment:
I'd probably call this currentRoute or something other than resolvedRoute. Otherwise we're going to have problems talking about which "resolved route"